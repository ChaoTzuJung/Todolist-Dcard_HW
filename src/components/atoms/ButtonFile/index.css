.input-file {
	position: fixed;
	top: 0;
	right: 0;
	opacity: 0;
	height: 0;
	width: 0;
}

.button-file {
	background: #c8c8c8;
	border-radius: 2px;
	padding: 8px;

	& > div {
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		width: 100%;
		height: 100%;
	}

	&.button-style {
		display: flex;
		justify-content: center;
		position: relative;
		height: 30px;
		padding: 0 10px 0 12px;
		border: solid 1px var(--black);
		border-radius: 0;
		outline: none;
		transition: all 0.3s;
		background: var(--white);

		& > div {
			display: flex;
			justify-content: center;
			align-items: center;
			text-align: center;
			font-size: 12px;
			line-height: 1.75;
			letter-spacing: 3.4px;
			color: var(--black);

			& > i {
				font-size: 16px;
				margin-right: 3px;
			}
		}

		& > .no-letter-spacing {
			letter-spacing: 0;
		}

		/* &:hover {
			cursor: pointer;
			background: var(--lightGray);
		}

		&:focus {
			border: 1px solid var(--accent);
		}

		&:active {
			border: 1px solid var(--accent);
			background: var(--pinkLight);
		} */
	}
}
