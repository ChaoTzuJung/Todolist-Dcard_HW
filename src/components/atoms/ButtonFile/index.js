import React, { Fragment, PureComponent } from 'react';
import classnames from 'classnames';

import styles from './index.css';

class ButtonFile extends PureComponent {
	constructor(props) {
		super(props);

		this.onClick = this.onClick.bind(this);
	}

	onClick(e) {
		this.refInput.click();
		const { onClick } = this.props;

		if (typeof onClick !== 'undefined') {
			onClick(e);
		}
	}

	render() {
		const { className, onChange, children, type, lan, ...other } = this.props;

		return (
			<Fragment>
				<button
					className={classnames(styles.buttonFile, className, {
						[styles.buttonStyle]: type === 'button',
					})}
					type="button"
					onClick={this.onClick}
					{...other}
				>
					<div>{children}</div>
				</button>
				<input
					ref={input => {
						this.refInput = input;
					}}
					className={styles.inputFile}
					type="file"
					onChange={onChange}
				/>
			</Fragment>
		);
	}
}

export default ButtonFile;
