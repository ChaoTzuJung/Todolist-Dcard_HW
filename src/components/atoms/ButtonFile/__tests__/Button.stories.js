import React from 'react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import ButtonFile from 'components/atoms/ButtonFile';

const stories = storiesOf('Commons|atoms/ButtonFile', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<ButtonFile>
		<i className="material-icons" style={{ color: '#FFFFFF' }}>
			add
		</i>
	</ButtonFile>
));

stories.add('with button style', () => <ButtonFile type="button">選擇檔案</ButtonFile>);
