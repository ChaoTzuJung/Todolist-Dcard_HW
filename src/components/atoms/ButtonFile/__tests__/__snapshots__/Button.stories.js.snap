// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<div
  style={
    Object {
      "margin": "60px",
    }
  }
>
  <ButtonFile>
    <button
      className="buttonFile"
      onClick={[Function]}
      type="button"
    >
      <div>
        <i
          className="material-icons"
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          add
        </i>
      </div>
    </button>
    <input
      className="inputFile"
      type="file"
    />
  </ButtonFile>
</div>
`;

exports[`with button style 1`] = `
<div
  style={
    Object {
      "margin": "60px",
    }
  }
>
  <ButtonFile
    type="button"
  >
    <button
      className="buttonFile buttonStyle"
      onClick={[Function]}
      type="button"
    >
      <div>
        選擇檔案
      </div>
    </button>
    <input
      className="inputFile"
      type="file"
    />
  </ButtonFile>
</div>
`;
